File: CLASS 2/11866 요세푸스 문제 0.cs
using System;

namespace ConsoleApp2
{
    public class Program
    {
        static void Main(string[] args)
        {
            string str = Console.ReadLine();
            string[] str_split = str.Split(' ');

            int N = int.Parse(str_split[0]);
            int K = int.Parse(str_split[1]);
            int[] nums = new int[1000];
            Console.Write("<");

            int num = 0;
            int index = 0;
            int print_count = 0;
            while (true)
            {
                int count = K;
                while (count > 0)
                {
                    num++;
                    if (num > N)
                        num = 1;

                    while (true)
                    {
                        bool flage = true;
                        for (int j = 0; j < index; j++)
                        {
                            if (num == nums[j])
                            {
                                num++;
                                if (num > N)
                                    num = 1;
                                flage = false;
                                break;
                            }
                        }

                        if (flage == true)
                            break;
                    }
                    count--;
                }
                nums[index++] = num;
                print_count++;
                Console.Write(num);
                if (print_count != N)
                    Console.Write(", ");
                if (print_count == N)
                {
                    Console.Write(">");
                    break;
                }
            }
        }
    }
}

File: CLASS 2/11866 요세푸스 문제 0.c
#define _CRT_SECURE_NO_WARNINGS // scanf사용시 에러무시
#include <stdio.h>


int main(void)
{
    int N;
    int K = 0;
    int num = 0;
    int nums[1000];

    scanf("%d %d", &N, &K);
    printf("<");

    int count = 0;
    int i = 0;
    int print_count = 0;
    while (1)
    {
        count = K;
        while (count > 0)
        {
            num++;
            if (num > N)
                num = 1;

            while (1)
            {
                int flage = 0;
                for (int j = 0; j < i; j++)
                {
                    if (num == nums[j])
                    {
                        flage = 1;
                        num++;
                        if (num > N)
                            num = 1;
                        break;
                    }
                }

                if (flage == 0)
                    break;
            }
            count--;
        }
        nums[i++] = num;
        print_count++;
        
        printf("%d", num);
        if (print_count != N)
            printf(", ");
        if (print_count == N)
        {
            printf(">");
            break;
        }
    }
    return (0);
}
